weblate/trans/models/projecttoken.py

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright © 2021 Christian Köberl
+#
+# This file is part of Weblate <https://weblate.org/>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+from django.db import models
+from django.utils.translation import gettext_lazy
+
+
+class ProjectToken(models.Model):
+    """Project token for API access."""
+
+    project = models.ForeignKey("Project", on_delete=models.deletion.CASCADE)
+    name = models.CharField(verbose_name=gettext_lazy("Name"), max_length=100)
+    token = models.CharField(
+        verbose_name=gettext_lazy("Token value"),
+        max_length=100,
+        help_text=gettext_lazy("Token value to use in API"),
```

Comment:
Perhaps provide a default value so that token is always initialized?

```suggestion
        help_text=gettext_lazy("Token value to use in API"),
        default=token_hex,
```

Reply:
Done that.